### ISSUE TITLE
there is a fault in the torch-baidu-ctc

### ISSUE THREAD

**Issue by firewood1996** [liecn/PyramidFL_mobicom22#1]
when I install the package torch-baidu-cc, there is a error. I can not find a solution about this error. Could you please tell me how to resolve?

**Comment by liecn** [comment]
I don't think it is necessary. You can skip it.

**Comment by liecn** [comment]
Ok. I realize we need it in the loss function. Could you please try "pip install torch-baidu-ctc"? Please show the logs if it doesn't work.
For your reference, https://pypi.org/project/torch-baidu-ctc/

**Comment by firewood1996** [comment]
thanks for your reply, actually I cannot install this package, maybe  due to torch-baidu-ctc related to the torch version 1.1.0. However, python3.7 cannot install this version torch. I will try python 3.6 again.

**Comment by firewood1996** [comment]
HI, I solved this problem due to replacing the torch-baidu-ctc with torch.nn.CTCLoss. But I still have another question, what is the main code? If I wangt to train a model, what is the main code I start with? Is learner.py? or parameter_sever.py? or manager.py?

**Comment by liecn** [comment]
> 
Great!

It's manager.py. Please see the instructions in README "Reproduce Results from Scratch".

**Comment by firewood1996** [comment]
Thanks！But I still have a question about what the function of parameter_server.py and learner.py?

**Comment by liecn** [comment]
You can see that we call the function of parameter_server.py and learner.py in manager.py. Thanks.

**Comment by liecn** [comment]
Resolved.
